lib/server-app.js

```diff
@@ -65,6 +65,29 @@ proto.middlewareFromConfig = function(factory, config) {
   }
 
   var handler = factory.apply(null, params);
+
+  // Check if methods/verbs filter exists
+  var verbs = config.methods || config.verbs;
+  if (Array.isArray(verbs)) {
+    if (verbs.indexOf('all') === -1) {
+      var originalHandler = handler;
+      if (handler.length <= 3) {
+        handler = function(req, res, next) {
+          if (verbs.indexOf(req.method.toLowerCase()) === -1) {
```

Comment:
Is it worth optimising this lookup (together with L83) and replace `Array.prototype.indexOf` with object-property lookup? 

I think it would be also good to normalise method/verbs that are coming from the JSON config (apply `toLowerCase`).

```js
var verbs = Object.create(null);
(config.methods || config.verbs).forEach(function(val) { verbs[val.toLowerCase()] = true; });
// later in the handler:
if (verbs[req.method.toLowerCase]) 
```